js/private/js_image_layer.mjs

```diff
@@ -1,9877 +1,339 @@
-function setCancelFlag(data) {
-    if (data.passiveListener != null) {
-        if (
-            typeof console !== "undefined" &&
-            typeof console.error === "function"
-        ) {
-            console.error(
-                "Unable to preventDefault inside passive event listener invocation.",
-                data.passiveListener
-            );
-        }
-        return
-    }
-    if (!data.event.cancelable) {
-        return
-    }
-
-    data.canceled = true;
-    if (typeof data.event.preventDefault === "function") {
-        data.event.preventDefault();
+function findKeyByValue(entries, value) {
+    const found = entries[value];
+    if (!found) {
+        return undefined
+    } else if (typeof found != "string") {
+        // matched against the real entry. 
+        return undefined
     }
+    return found
 }
 
-/**
- * @see https://dom.spec.whatwg.org/#interface-event
- * @private
- */
-/**
- * The event wrapper.
- * @constructor
- * @param {EventTarget} eventTarget The event target of this dispatching.
- * @param {Event|{type:string}} event The original event to wrap.
- */
-function Event(eventTarget, event) {
-    privateData.set(this, {
-        eventTarget,
-        event,
-        eventPhase: 2,
-        currentTarget: eventTarget,
-        canceled: false,
-        stopped: false,
-        immediateStopped: false,
-        passiveListener: null,
-        timeStamp: event.timeStamp || Date.now(),
-    });
-
-    // https://heycam.github.io/webidl/#Unforgeable
-    Object.defineProperty(this, "isTrusted", { value: false, enumerable: true });
-
-    // Define accessors
-    const keys = Object.keys(event);
-    for (let i = 0; i < keys.length; ++i) {
-        const key = keys[i];
-        if (!(key in this)) {
-            Object.defineProperty(this, key, defineRedirectDescriptor(key));
+async function readlinkSafe(p) {
+    try {
+        const link = await readlink(p)
+        return path.resolve(path.dirname(p), link)
+    } catch (e) {
+        if (e.code == 'EINVAL') {
+            return p
         }
+        if (e.code == 'ENOENT') {
+            // That is as far as we can follow this symlink in this layer so we can only
+            // assume the file exists in another layer
+            return p
+        }
+        throw e
     }
 }
 
-// Should be enumerable, but class methods are not enumerable.
-Event.prototype = {
-    /**
-     * The type of this event.
-     * @type {string}
-     */
-    get type() {
-        return pd(this).event.type
-    },
+const EXECROOT = process.cwd();
 
-    /**
-     * The target of this event.
-     * @type {EventTarget}
-     */
-    get target() {
-        return pd(this).eventTarget
-    },
-
-    /**
-     * The target of this event.
-     * @type {EventTarget}
-     */
-    get currentTarget() {
-        return pd(this).currentTarget
-    },
+// Resolve symlinks while staying inside the sandbox.
+async function resolveSymlink(p) {
+    let prevHop = path.resolve(p)
```

Comment:
I think the `path.resolve` could be dropped?

Reply:
i don't remember why it was put there in the first place, so i will just leave it as is.